src/Raven.Client/Document/DocumentStore.cs

```diff
@@ -278,11 +278,10 @@ public override IDocumentSession OpenSession(OpenSessionOptions options)
             currentSessionId = sessionId;
             try
             {
-                var session = new DocumentSession(options.Database, this, Listeners, sessionId,
-                    SetupCommands(DatabaseCommands, options.Database, options.Credentials, options))
-                    {
-                        DatabaseName = options.Database ?? DefaultDatabase ?? MultiDatabase.GetDatabaseName(Url)
-                    };
+                var databaseName = options.Database ?? DefaultDatabase ?? MultiDatabase.GetDatabaseName(Url);
+                var requestExecuter = _requestExecuters.GetOrAdd(databaseName, new RequestExecuter(Url, databaseName, ApiKey));
```

Comment:
Here you are creating a request executer every single time.
Better to have Lazy<RequestExecuter> and create that via a delegate